tools/webpack/packages.js

```diff
@@ -141,6 +141,20 @@ module.exports = {
 			return `webpack://${ info.namespace }/${ info.resourcePath }`;
 		},
 	},
+	module: {
+		rules: [
+			...baseConfig.module.rules,
+			{
+				test: /\.wasm$/,
+				type: 'asset/resource',
+				generator: {
+					// FIXME: Do not hardcode path.
+					filename: './build/vips/[name].wasm',
+					publicPath: '',
+				},
+			},
+		],
+	},
```

Comment:
It's configured as a webpack "asset", similar to images and other assets. Webpack basically copies the file and exposes the path when the asset is imported.

It shouldn't be published to npm for now, so I don't think that's a concern at this time.

For WordPress scripts it should work because we can figure out the path that needs to be used to access the wasm assets.

These wasm assets are one reason I'd really like to use script modules for this package. Webpack has some wasm module support and the library relies on some modules features, so there may be some simplification available.

Reply:
> It shouldn't be published to npm for now, so I don't think that's a concern at this time.

For me that's a concern, my guess is that we're going to add the `vips` packages as a dependency of `block-editor` package. So my question is what happens to third-party block editors, will they have to change how their applications builds, feels heavy? How do regular npm packages (in the npm registry) that use wasm dependencies work.

Reply:
In other words, will we have to update https://wordpress.org/gutenberg-framework/ as a consequence of this?

Reply:
> For me that's a concern, my guess is that we're going to add the vips packages as a dependency of block-editor package.

Indirectly through the new `upload-media` package, yes. Note that first this will be behind an experiment in the plugin only, and wouldn't be exposed through the published npm packages (at least that's my idea). So we have some time to iterate on this. Right now this change doesn't affect any published packages.

> So my question is what happens to third-party block editors, will they have to change how their applications builds, feels heavy? How do regular npm packages (in the npm registry) that use wasm dependencies work.
> In other words, will we have to update https://wordpress.org/gutenberg-framework/ as a consequence of this?

My understanding is that third-party block editor applications will need to update their bundler config to ensure that it supports wasm files. Once this package is published as a script module only, usage might become a bit easier.

Vite (as used by https://wordpress.org/gutenberg-framework/) has some wasm support built-in, but for ESM integration there is `vite-plugin-wasm`

Reply:
I wonder if `asset/inline` is actually better for us, that way we can ship npm packages that are ready to use regardless of the bundler config. I'm not sure how I feel about pushing this burden towards consumers. 

Ok for me to start as a Gutenberg experiment not exposed to WordPress and npm users.

Reply:
> I wonder if `asset/inline` is actually better for us, that way we can ship npm packages that are ready to use regardless of the bundler config.

That sounds really bad for performance if we shove several MB worth of WASM files as data URIs into the JS bundles... I'd much rather have them as separate files so that they can be cached by the browser or even offloaded to a CDN.

Reply:
> That sounds really bad for performance if we shove several MB worth of WASM files as data URIs into the JS bundles... I'd much rather have them as separate files so that they can be cached by the browser or even offloaded to a CDN.

Isn't it the same thing for the JS, it can also be cached and offloaded if needed, not sure I follow what's different here.